.github/workflows/main.yaml

```diff
@@ -0,0 +1,44 @@
+---
+name: Continuous integration
+
+on:
+  push:
+  pull_request:
+
+env:
+  HAS_SECRETS: ${{ secrets.HAS_SECRETS }}
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+    name: Continuous integration
+    timeout-minutes: 10
+#    if: "!startsWith(github.event.head_commit.message, '[skip ci] ')"
```

Comment:
I'll _uncomment_ this. 
it is nice to have some way to skip the CI.
I disabled it because at some point I wanted to test some local github executors like https://github.com/nektos/act  before committing the CI to GH. but it is more trouble than its worth.